server/routes/purchase.js

```diff
@@ -0,0 +1,25 @@
+exports = module.exports = function (app) {
+  const express = require('express');
+  const queue = require('express-queue');
+  const router = express.Router();
+  const { Ticket } = app.db.models;
+
+  router.put('/:ticket', async (req, res) => {
+    try {
+      const { params: { ticket }, body: { amount } } = req;
+      const data = await Ticket.findOne({ _id: ticket });
```

Comment:
In real world, the requests will be arrived in parallel from different thread, worker, that's said this view will be called in different thread so await here could not pause the incoming requests on other thread or worker.